plugins/services/src/js/reducers/serviceForm/MultiContainerHealthChecks.js

```diff
@@ -3,79 +3,423 @@ import {
-      if (joinedPath === 'healthChecks') {
-        switch (type) {
-          case ADD_ITEM:
-            this.healthChecks.push({protocol: 'COMMAND'});
-            break;
-          case REMOVE_ITEM:
-            this.healthChecks = this.healthChecks.filter((item, index) => {
-              return index !== value;
-            });
-            break;
-        }
+function reduceHttpHealthCheck(state, field, value) {
+  const newState = Object.assign({}, state);
+  newState.http = Object.assign({}, newState.http || {});
 
-        return this.healthChecks.map(mapHealthChecks);
+  switch (field) {
+    case 'endpoint':
+      newState.http.endpoint = value;
+      break;
+
+    case 'path':
+      newState.http.path = value;
+      break;
+
+    case 'https':
+      if (value) {
+        newState.http.scheme = HTTPS;
+      } else {
+        newState.http.scheme = null;
       }
+      break;
+  }
 
-      const index = joinedPath.match(/\d+/)[0];
-      if (type === SET) {
-        if (`healthChecks.${index}.command` === joinedPath) {
-          this.healthChecks[index].command = value;
-        }
-        if (`healthChecks.${index}.path` === joinedPath) {
-          this.healthChecks[index].path = value;
-        }
-        if (`healthChecks.${index}.gracePeriodSeconds` === joinedPath) {
-          this.healthChecks[index].gracePeriodSeconds = parseInt(value, 10);
+  return newState;
+}
+
+function reduceFormHttpHealthCheck(state, field, value) {
+  const newState = Object.assign({}, state);
+  newState.http = Object.assign({}, newState.http || {});
+
+  switch (field) {
+    case 'https':
+      newState.http.https = value;
+      break;
+  }
+
+  return newState;
+}
+
+/**
+ * JSON Parser Fragment for `TcpHealthCheck` type
+ *
+ * @param {Object} healthCheck - The healthcheck data to parse
+ * @param {Array} path - The path prefix to the transaction
+ * @returns {Array} - Returns an array with the new transactions to append
+ */
+function parseTcpHealthCheck(healthCheck, path) {
+  const memo = [];
+
+  if (healthCheck.endpoint != null) {
+    memo.push(new Transaction(
+      path.concat(['endpoint']),
+      healthCheck.endpoint,
+      SET
+    ));
+  }
+
+  return memo;
+}
+
+function reduceTcpHealthCheck(state, field, value) {
+  const newState = Object.assign({}, state);
+  newState.tcp = Object.assign({}, newState.tcp || {});
+
+  switch (field) {
+    case 'endpoint':
+      newState.tcp.endpoint = value;
+      break;
+  }
+
+  return newState;
+}
+
+/**
+ * JSON Parser Fragment for `CommandHealthCheck` type
+ *
+ * @param {Object} healthCheck - The healthcheck data to parse
+ * @param {Array} path - The path prefix to the transaction
+ * @returns {Array} - Returns an array with the new transactions to append
+ */
+function parseCommandHealthCheck(healthCheck, path) {
+  const memo = [];
+  const {command={}} = healthCheck;
+
+  if (command.shell != null) {
+    memo.push(new Transaction(
+      path.concat(['command', 'shell']),
```

Comment:
Let's rename this to `command.type` to make obvious that this is toggling the command type